### PR TITLE
Adds GA4 custom event product_download to /all

### PR DESCRIPTION
**Pull request by stephaniehobson**
## One-line summary

Adds GA4 custom event product_download and adds event tracking to downloads on /all.

## Significant changes and points to review

Adds datalayer-product-download.es6.js and converts all-downloads-unified-ini to es6 to allow importing it.

## Issue / Bugzilla link

https://github.com/mozilla/bedrock/issues/13238

## Testing

https://www-demo4.allizom.org/en-US/firefox/all/ 

To test this work:

Bedrock code:
- [x] new tests provide adequate coverage
- [x] good checks exist so things don't break if the data is missing or malformed
- [x] `product_download` event is fired for Firefox download
- [x] `product_download` event is fired for Firefox Beta download
- [x] `product_download` event is fired for Firefox Developer download
- [x] `product_download` event is fired for Firefox Nightly download
- [x] `product_download` event is fired for Firefox ESR download
- [x] `product_download` event is fired for Firefox Android Play Store button
- [x] `product_download` event is fired for Firefox Android Beta Play Store button
- [x] `product_download` event is fired for Firefox Android Nightly Play Store button
- [x] `product_download` event is fired for Firefox iOS App Store button

GTM:
- Switch to the workspace in the www.mozilla.org workspace called "GA4 product_download"
- Click "preview" to open the tag assistant website.
- Add the domain https://www-demo4.allizom.org/en-US/firefox/all/ and click it 
- Disable add-ons and toggle off DNT, etc until tag assistant indicates it has a connection
- Navigate to the /all page and start clicking the download/store buttons
- Return to the tag assistant window and look through the 'product_download' events to verify they have the correct product, platform, release_channel, and download_language.
- [x] Firefox download
- [x] Firefox Beta download
- [x] Firefox Developer download
- [ ] Firefox Nightly download
- [ ] Firefox ESR download
- [ ] Firefox Android Play Store button **should not identify a download_language
- [ ] Firefox Android Beta Play Store button **should not identify a download_language
- [ ] Firefox Android Nightly Play Store button **should not identify a download_language
- [x] Firefox iOS App Store button *should not identify a release_channel **should not identify a download_language

GA4:
- Switch to the **www.mozilla.org - GA4 Testing** property
- Click the Admin menu item (bottom left)
- Click the DebugView link (last item in the second column before the colourful product links)
- Your tag assistant session should already have sent data here, you might have to pick a device out of the drop down until "Debug Device"
- [x] Verify the product_download events have been logged

Tests:
- [x] unit tests pass
- [x] integration tests pass

### CODE REVIEW

File: media/css/firefox/new/desktop/download.scss [hunk 1]
```diff
@@ -936,7 +936,7 @@ button.mzp-c-cta-link {
         @include text-body-sm;
         @include bidi((
             (border-left, 1px solid $color-marketing-gray-20, border-right, 0),
-            (text-align, center, center), // needs higher specificity for RTL
```
Comment (stephaniehobson) [inline media/css/firefox/new/desktop/download.scss:939]: Caught when I ran `npm format` not related to this work. I guess I could pull it out if you want.

File: tests/unit/spec/base/datalayer-begincheckout.js [hunk 2]
```diff
@@ -14,18 +14,20 @@ import TrackBeginCheckout from '../../../../media/js/base/datalayer-begincheckou
...
-            {
-                item_id: 'testid',
-                item_name: 'email',
-                item_category: 'relay',
-                item_variant: 'yearly',
-                price: '11.88',
-                discount: '12.00'
-            }
-        ]
+        ecommerce: {
```
Comment (stephaniehobson) [inline tests/unit/spec/base/datalayer-begincheckout.js:17]: Turned up a bug in this test while adding these ones. Also submitted as https://github.com/mozilla/bedrock/pull/13237

**Comment by stephaniehobson** [comment]
Integration tests timed out on an unrelated test. I'll push again in the morning.

**Comment by stephaniehobson** [comment]
Integration tests passed.

**Review by nathan-barrett — APPROVED** [review]
r+ great work here @stephaniehobson!!

**Pull request merged**